src/dmd/statementsem.d

```diff
@@ -4561,6 +4565,397 @@ Statement scopeCode(Statement statement, Scope* sc, Statement* sentry, Statement
+            {
+                return null;
+            }
+            sc = sc.push(fs.sym);
+            auto a = fs.statement.flatten(sc);
+            sc = sc.pop();
+            if (!a)
+            {
+                return a;
+            }
+            auto b = new Statements(a.dim);
+            foreach (i, s; *a)
+            {
+                (*b)[i] = s ? new ForwardingStatement(s.loc, fs.sym, s) : null;
+            }
+            return b;
+
+        case STMT.Conditional:
+            auto cs = statement.isConditionalStatement();
+            Statement s;
+
+            //printf("ConditionalStatement::flatten()\n");
+            if (cs.condition.include(sc))
+            {
+                DebugCondition dc = cs.condition.isDebugCondition();
+                if (dc)
+                {
+                    s = new DebugStatement(cs.loc, cs.ifbody);
+                    debugThrowWalker(cs.ifbody);
+                }
+                else
+                    s = cs.ifbody;
+            }
+            else
+                s = cs.elsebody;
+
+            auto a = new Statements();
+            a.push(s);
+            return a;
+
+        case STMT.StaticForeach:
+            auto sfs = statement.isStaticForeachStatement();
+            sfs.sfe.prepare(sc);
+            if (sfs.sfe.ready())
+            {
+                auto s = makeTupleForeach!(true, false)(sc, sfs.sfe.aggrfe, sfs.sfe.needExpansion);
+                auto result = s.flatten(sc);
+                if (result)
+                {
+                    return result;
+                }
+                result = new Statements();
+                result.push(s);
+                return result;
+            }
+            else
+            {
+                auto result = new Statements();
+                result.push(new ErrorStatement());
+                return result;
+            }
+
+        case STMT.Debug:
+            auto ds = statement.isDebugStatement();
+            Statements* a = ds.statement ? ds.statement.flatten(sc) : null;
+            if (a)
+            {
+                foreach (ref s; *a)
+                {
+                    s = new DebugStatement(ds.loc, s);
+                }
+            }
+            return a;
+
+        case STMT.Label:
+            auto ls = statement.isLabelStatement();
+            Statements* a = null;
+            if (ls.statement)
+            {
+                a = ls.statement.flatten(sc);
+                if (a)
+                {
+                    if (!a.dim)
+                    {
+                        a.push(new ExpStatement(ls.loc, cast(Expression)null));
+                    }
+
+                    // reuse 'this' LabelStatement
+                    ls.statement = (*a)[0];
+                    (*a)[0] = ls;
+                }
+            }
+            return a;
+
+        case STMT.Compile:
+            auto cs = statement.isCompileStatement();
+
+            auto errorStatements()
```

Comment:
there are other cases that use a similar pattern. Move this outside the `switch`